### ISSUE TITLE
CLI applications may also require Tor

### ISSUE THREAD

**Issue by nns52k** [Foundation-Devices/tor#40]
Is it possible to remove flutter from dependency list of this package tor? If "yes", CLI programs could use package tor too.

**Comment by sneurlax** [comment]
Yes, Dart 3.6's Native Assets feature allows the Flutter-based irondash/cargokit used for integrating cargo builds into flutter builds can be replaced with the newer irondash/native_toolchain_rust.  We can re-architect to wrap the same underlying Rust crate with a new Dart-only build system and thus remove Flutter as a dependency.

This is what my package https://pub.dev/packages/arti does.  Sorry to advertise here, icota and Foundation-Devices, but once Dart releases Native Assets in a stable version then we can replace cargokit with native_toolchain_rust for this package.
